api/data.go

```diff
@@ -79,17 +82,37 @@ func (api *API) CreateOrUpdateCacheTime(ctx context.Context, w http.ResponseWrit
 		ID: id,
 	}
 
-	err := json.NewDecoder(req.Body).Decode(&docToInsertOrUpdate)
+	// Check request body not empty
+	if req.ContentLength == 0 {
+		log.Error(ctx, "createOrUpdateCacheTime endpoint: empty request body", errors.New("empty request body"))
+		http.Error(w, "bad request: empty request body", http.StatusBadRequest)
+		return
+	}
+
+	decoder := json.NewDecoder(req.Body)
+	decoder.DisallowUnknownFields() // disallow unknown fields in the request body
+
+	err := decoder.Decode(&docToInsertOrUpdate)
+	if err != nil {
+		// Handle error for unknown fields, incorrect field type and decode
+		log.Error(ctx, "createOrUpdateCacheTime endpoint:", err)
+		http.Error(w, fmt.Errorf("bad request: %v", err).Error(), http.StatusBadRequest)
```

Comment:
Given that this API is only used internally, this is not an issue, but just bear in mind that by simply forwarding the decoder errors you're exposing implementation details to the user, which is a (small) vulnerability. In this case, given that this is an internal API that's also open source, it's not a big deal, but just have that in mind in the future when designing other APIs.